src/ipred.rs

```diff
@@ -243,21 +243,20 @@ unsafe fn cfl_pred<BD: BitDepth>(
     width: c_int,
     height: c_int,
     dc: c_int,
-    mut ac: *const i16,
+    ac: &[i16; 1024],
     alpha: c_int,
     bd: BD,
 ) {
-    let mut y = 0;
-    while y < height {
-        let mut x = 0;
-        while x < width {
-            let diff = alpha * *ac.offset(x as isize) as c_int;
-            *dst.offset(x as isize) = bd.iclip_pixel(dc + apply_sign(diff.abs() + 32 >> 6, diff));
-            x += 1;
+    let mut ac = ac.as_slice();
```

Comment:
```suggestion
    let mut ac = &ac[..width * height];
```

This should do all the bounds checks on `ac` at once and elide them elsewhere.